### PR TITLE
Add Documentation for delete global_pool

### PR DESCRIPTION
**Pull request by Kandarimegha**
## Description
In this commit:
Bug Fixes
Documentation Update:
- Included  document for deleting a Global IPPool.

SLAAC Support Parameter:
- Made slaacSupport an optional parameter by removing the default value of True.

IPv6 Address Normalization:
- Implemented a mechanism to normalize IPv6 addresses for consistent comparison.

- Modify the code to check all the invalid global pool name at once.

## Type of Change
- [x] Bug fix
- [ ] New feature
- [ ] Breaking change
- [ ] Documentation update

## Checklist
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] All the sanity checks have been completed and the sanity test cases have been executed

## Ansible Best Practices
- [x] Tasks are idempotent (can be run multiple times without changing state)
- [ ] Variables and secrets are handled securely (e.g., using `ansible-vault` or environment variables)
- [x] Playbooks are modular and reusable
- [ ] Handlers are used for actions that need to run on change

## Documentation
- [x] All options and parameters are documented clearly.
- [x] Examples are provided and tested.
- [x] Notes and limitations are clearly stated.

### CODE REVIEW

File: plugins/module_utils/dnac.py [hunk 1]
```diff
@@ -2160,13 +2160,26 @@ def compare_list(list1, list2):
 def fn_comp_key(k, dict1, dict2):
     return dnac_compare_equality(dict1.get(k), dict2.get(k))
 
+def normalize_ipv6_address(ipv6):
+    """
+    Normalize an IPv6 address for consistent comparison.
+    """
+    try:
+        return str(ipaddress.IPv6Address(ipv6))
+    except ValueError:
```
Comment (madhansansel) [inline plugins/module_utils/dnac.py:2169]: Can we print a log message?

        self.log("Invalid IPv6 address: {}".format(ipv6))

File: plugins/module_utils/dnac.py [hunk 2]
```diff
@@ -2160,13 +2160,26 @@ def compare_list(list1, list2):
 def fn_comp_key(k, dict1, dict2):
     return dnac_compare_equality(dict1.get(k), dict2.get(k))
 
+def normalize_ipv6_address(ipv6):
+    """
+    Normalize an IPv6 address for consistent comparison.
+    """
+    try:
```
Comment (madhansansel) [inline plugins/module_utils/dnac.py:2167]: if not isinstance(ipv6, str):
        raise TypeError("Input must be a string representing an IPv6 address.")

    try:
        normalized_address = str(ipaddress.IPv6Address(ipv6))
        return normalized_address

File: plugins/modules/network_settings_workflow_manager.py [hunk 3]
```diff
@@ -1310,9 +1329,13 @@ def get_aaa_settings_for_site(self, site_name, site_id):
             network_aaa = aaa_network_response.get("response", {}).get("aaaNetwork")
             client_and_endpoint_aaa = aaa_network_response.get("response", {}).get("aaaClient")
 
-            if not network_aaa or not client_and_endpoint_aaa:
-                self.log("No AAA settings found for site '{0}' (ID: {1})".format(site_name, site_id), "WARNING")
-                return None, None
+            if network_aaa or not client_and_endpoint_aaa:
```
Comment (madhansansel) [inline plugins/modules/network_settings_workflow_manager.py:1332]: if network_aaa is there, then we return without checking the next condition. Is that correct?

File: plugins/module_utils/dnac.py (hunk 1 above)
Comment (Kandarimegha) [reply plugins/module_utils/dnac.py:2173]: Done

File: plugins/module_utils/dnac.py (hunk 2 above)
Comment (Kandarimegha) [reply plugins/module_utils/dnac.py:2170]: modified

File: plugins/modules/network_settings_workflow_manager.py (hunk 3 above)
Comment (Kandarimegha) [reply plugins/modules/network_settings_workflow_manager.py]: Yes, it should check both condition. so replaced "or" with "and".

File: plugins/modules/network_settings_workflow_manager.py [hunk 4]
```diff
@@ -1310,9 +1329,13 @@ def get_aaa_settings_for_site(self, site_name, site_id):
             network_aaa = aaa_network_response.get("response", {}).get("aaaNetwork")
```
[the remark this answers is not in the document]
Comment (Kandarimegha) [reply plugins/modules/network_settings_workflow_manager.py]: done the code optimization

**Review by madhansansel — APPROVED** [review]

**Comment by madhansansel** [comment]
Having sanity issues. please fix it.

ERROR: Found 4 pep8 issue(s) which need to be resolved:
ERROR: plugins/module_utils/dnac.py:2163:1: E302: expected 2 blank lines, found 1
ERROR: plugins/module_utils/dnac.py:2174:55: W291: trailing whitespace
ERROR: plugins/module_utils/dnac.py:[217](https://github.com/cisco-en-programmability/catalyst-center-ansible-intg/actions/runs/12166799891/job/33933866430?pr=47#step:8:218)7:1: E302: expected 2 blank lines, found 1
ERROR: plugins/module_utils/dnac.py:2185:17: E117: over-indented
See documentation for help: https://docs.ansible.com/ansible-core/devel/dev_guide/testing/sanity/pep8.html
Running sanity test "pslint"
No tests applicable.
Running sanity test "pylint"
Run command: /root/.ansible/test/venv/sanity.pylint/3.13/4d265355/bin/python /root/ansible/test/lib/ansible_test/_util/controller/tools/collection_detail.py /root/ansible_collections/cisco/dnac
Checking 1 file(s) in context 'module_utils' (target) with config: /root/ansible/test/lib/ansible_test/_util/controller/sanity/pylint/config/collection.cfg
Run command: /root/.ansible/test/venv/sanity.pylint/3.13/4d265355/bin/python -m pylint --py-version 3.8 --load-plugins deprecated,hide_unraisable,string_format,unwanted --rcfile /root/ansible/test/lib/ansible_test/_util/controller/sanity/pylint/config/collection.cfg --jobs 0 --reports n --output-format json --collection-name cisco.dnac --collection-version 6.25.0 plugins/module_utils/dnac.py
ERROR: Found 2 pylint issue(s) which need to be resolved:
ERROR: plugins/module_utils/dnac.py:2174:8: ansible-bad-function: Call module.log or module.debug instead of builtins.print
ERROR: plugins/module_utils/dnac.py:2174:54: trailing-whitespace: Trailing whitespace

**Comment by madhansansel** [comment]
Please kindly fix the CI/CD pipeline

**Pull request merged**